crd/overlayextensionconfig/client.go

```diff
@@ -0,0 +1,105 @@
+package overlayextensionconfig
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/Azure/azure-container-networking/crd"
+	"github.com/Azure/azure-container-networking/crd/overlayextensionconfig/api/v1alpha1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	typedv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Scheme is a runtime scheme containing the client-go scheme and the OverlayExtensionConfig scheme.
+var Scheme = runtime.NewScheme()
+
+func init() {
+	_ = scheme.AddToScheme(Scheme)
+	_ = v1alpha1.AddToScheme(Scheme)
+}
+
+// Installer provides methods to manage the lifecycle of the OverlayExtensionConfig resource definition.
+type Installer struct {
+	cli typedv1.CustomResourceDefinitionInterface
+}
+
+func NewInstaller(c *rest.Config) (*Installer, error) {
+	cli, err := crd.NewCRDClientFromConfig(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init crd client")
+	}
+	return &Installer{
+		cli: cli,
+	}, nil
+}
+
+func (i *Installer) create(ctx context.Context, res *v1.CustomResourceDefinition) (*v1.CustomResourceDefinition, error) {
+	res, err := i.cli.Create(ctx, res, metav1.CreateOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create oec crd")
+	}
+	return res, nil
+}
+
+// Install installs the embedded OverlayExtensionConfig CRD definition in the cluster.
+func (i *Installer) Install(ctx context.Context) (*v1.CustomResourceDefinition, error) {
```

Comment:
I am not sure if `Install` method on any of the CRDs is used at all, while `InstallOrUpdate` method is used in Networking-Aquarius repo in [/src/dnc/requestcontroller/clustermanager](https://msazure.visualstudio.com/One/_git/Networking-Aquarius?path=/src/dnc/requestcontroller/clustermanager.go&version=GBmaster&line=125&lineEnd=136&lineStartColumn=1&lineEndColumn=3&lineStyle=plain&_a=contents) and in [src/dnc/requestcontroller/kubernetes/clustermanagerbuilder.go](https://msazure.visualstudio.com/One/_git/Networking-Aquarius?path=/src/dnc/requestcontroller/kubernetes/clustermanagerbuilder.go&version=GBmaster&line=185&lineEnd=200&lineStartColumn=1&lineEndColumn=3&lineStyle=plain&_a=contents)

Reply:
that would make sense we just want it in regardless of if we're installing or updating an existing one.

Reply:
AH this is not genrated.

Reply:
Strong proponent of only adding code you intend to use and adding more later, I would just remove this

Reply:
Removed `Install` method